### PR TITLE
Resurrect 'make test' on the JVM backend

### PR DESCRIPTION
**Pull request by usev6**
There are a lot of failing tests, but at least one *could* run 'make test' again.

### CODE REVIEW

**Comment by usev6** [comment]
I'm not sure if my changes to the Makefile templates are sane. I don't fully understand the used macros. But on my box (FreeBSD) the result looked fine for MoarVM and JVM.

**Comment by usev6** [comment]
Btw, this is part of working on https://github.com/rakudo/rakudo/issues/4191.

File: tools/templates/Makefile-backend-common.in
Comment (vrurg) [inline tools/templates/Makefile-backend-common.in:139]: `@nfpl()@` doesn't make any sense here because what it gets as input is `$(<backend_prefix>_BACKEND_SPECIFIC_TEST_FILES)`, i.e. a Makefile variable. See the comment above how to get it fixed.

File: tools/templates/js/Makefile.in
Comment (vrurg) [inline tools/templates/js/Makefile.in:25]: `nfp` again. But I wonder what does it supposed to do in first place?

File: tools/templates/jvm/Makefile.in
Comment (vrurg) [inline tools/templates/jvm/Makefile.in:44]: Same as above.

File: tools/templates/moar/Makefile.in
Comment (vrurg) [inline tools/templates/moar/Makefile.in:108]: And here too.

**Review by vrurg — CHANGES_REQUESTED** [review]

**Comment by vrurg** [comment]
> I'm not sure if my changes to the Makefile templates are sane. I don't fully understand the used macros. But on my box (FreeBSD) the result looked fine for MoarVM and JVM.

It's fine overall. Hope my notes will help get it straighten out.

**Comment by usev6** [comment]
Thanks for the helpful comments! I've pushed a commit that hopefully addresses the shortcomings you mentioned: https://github.com/rakudo/rakudo/pull/4195/commits/6556a1cd18. Before an eventual merge I'd squash that commit with the previous one, but I thought it might be clearer to not rewrite history for now.

Also I thought it would make sense to reuse the new Makefile variables for m-coretest6, too.

File: tools/templates/js/Makefile.in
Comment (usev6) [reply tools/templates/js/Makefile.in]: I'm not sure, I understand your question. Does it refer to introducing a new Makefile variable or to the use of ```@bpv()@```?

I thought it would make sense to declare the backend specific test dirs (sic!) in the backend specific Makefile templates. So a new variable looked logical to me.

Wrt ```@bpv()@```: The macro stuff looked scary to me, so I tried to cargo cult from the surrounding code. (I probably should have looked at ```NQP::Macros.pm```.) With my newly pushed commit I've changed to ```@bsv()@```, because we use that for ```METAMODEL_SOURCES``` (which looks structurally similar to the test dirs).

Comment (vrurg) [reply tools/templates/js/Makefile.in]: I'm sorry for the totally legit confusion of yours and recall my question. I overlooked the fact that the "duplicated" lines are from different backend templates.

**Review by vrurg — APPROVED** [review]
Looks good to me. Thank you!

**Comment by usev6** [comment]
That was just a force push with two commits squashed together (no other change). I'll merge once the checks are green.

**Pull request merged**
